Switch to O prevew2 sdk with changes in media

dakegu opened the pull request:
1. Switch to use PlaybackTransportControlGlue + VideoFragment

2. added seek demo for video.

3. Switch music to use PlaybackFragment. Add next() and previous()
in music example, remove FF/RW since the five seconds jump model
isn't matching PlaybackControlGlue.

A lot of code clean up.

dakegu2 committed:
Switch to O prevew2 sdk with changes in media

1. Switch to use PlaybackTransportControlGlue + VideoFragment

2. added seek demo for video.

3. Switch music to use PlaybackFragment. Add next() and previous()
in music example, remove FF/RW since the five seconds jump model
isn't matching PlaybackControlGlue.

A lot of code clean up.

WeiHsinChen commented on app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:
```diff
@@ -150,63 +105,56 @@ public void setupControlsRowPresenter(PlaybackControlsRowPresenter presenter) {
-            mRepeatAction.nextIndex();
-        } else if (action instanceof PlaybackControlsRow.ThumbsUpAction) {
+        if (action instanceof PlaybackControlsRow.ShuffleAction
+                ||action instanceof PlaybackControlsRow.RepeatAction) {
```
space after ||

WeiHsinChen commented on app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:
```diff
@@ -150,63 +105,56 @@ public void setupControlsRowPresenter(PlaybackControlsRowPresenter presenter) {
-            mRepeatAction.nextIndex();
-        } else if (action instanceof PlaybackControlsRow.ThumbsUpAction) {
+        if (action instanceof PlaybackControlsRow.ShuffleAction
+                ||action instanceof PlaybackControlsRow.RepeatAction) {
```
space after ||

dakegu replied:
done

WeiHsinChen commented on app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoExampleActivity.java:
```diff
@@ -35,12 +35,9 @@ public void onCreate(Bundle savedInstanceState) {
-        ft1.replace(R.id.videoFragment, new VideoSurfaceFragment(), VideoSurfaceFragment.TAG);
-        ft1.commit();
-
         FragmentTransaction ft2 = getFragmentManager().beginTransaction();
```
ft would be better

dakegu replied:
done

WeiHsinChen commented on app/src/main/res/drawable/ic_favorite_filled_24dp.xml:
```diff
@@ -5,5 +5,5 @@
     <path
         android:pathData="M12,21.35l-1.45,-1.32C5.4,15.36 2,12.28 2,8.5 2,5.42 4.42,3 7.5,3c1.74,0 3.41,0.81 4.5,2.09C13.09,3.81 14.76,3 16.5,3 19.58,3 22,5.42 22,8.5c0,3.78 -3.4,6.86 -8.55,11.54L12,21.35z"
-        android:fillColor="@color/song_row_favorite_color"/>
+        android:fillColor="#FF6E40"/>
```
It seems that color/song_row_favorite_color is still "#FF6E40" in this CL. Am I wrong?

dakegu replied:
it wont compile with new sdk

WeiHsinChen commented on app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicMediaPlayerGlue.java:
```diff
@@ -394,44 +195,69 @@ private void queueNextRefresh() {
     }
 
-    @Override protected void pausePlayback() {
+    @Override public void pause() {
```
Is there any reason to put Override with the method name in the same line?

dakegu replied:
All current methods are doing this way.  Might clean up later

WeiHsinChen commented on app/build.gradle:
```diff
@@ -1,13 +1,13 @@
-    compileSdkVersion 25
-    buildToolsVersion "25"
+    compileSdkVersion 'android-O'
+    buildToolsVersion '26.0.0-rc2'
```
Is '26.0.0-rc' already available?

dakegu2 committed:
code cleanup

dakegu2 committed:
code clean up 2

WeiHsinChen reviewed (approved):
LGTM +1. Lets ship it :shipit:!

        /\/\

                        /\/\
         O  O  O
                 O
                __|__
                || ||_____
                || ||    |
     --------------------------
      \   O   O   O   O      /
 ~ ~ ~ ~ ~ ~ ~ ~  ~ ~ ~ ~ ~ ~ ~ ~
  ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~

dakegu2 committed:
switch to google maven repository

since 26.0.0 beta1, android support library
switches to use google maven repository.
